data-plane/dispatcher/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/impl/http/WebClientCloudEventSender.java

```diff
@@ -43,76 +44,96 @@ public final class WebClientCloudEventSender implements CloudEventSender {
+   * @param egress egress configuration
    */
-  public WebClientCloudEventSender(final WebClient client,
-                                   final CircuitBreaker circuitBreaker,
-                                   final CircuitBreakerOptions circuitBreakerOptions,
-                                   final String target) {
-    this.circuitBreakerOptions = circuitBreakerOptions;
+  public WebClientCloudEventSender(final Vertx vertx,
+                                   final WebClient client,
+                                   final String target,
+                                   final DataPlaneContract.EgressConfig egress) {
+    Objects.requireNonNull(vertx);
     Objects.requireNonNull(client, "provide client");
     if (target == null || target.equals("")) {
       throw new IllegalArgumentException("provide a target");
     }
     if (!URI.create(target).isAbsolute()) {
       throw new IllegalArgumentException("provide a valid target, provided target: " + target);
     }
-    Objects.requireNonNull(circuitBreaker, "provide circuitBreaker");
 
+    this.vertx = vertx;
     this.client = client;
     this.target = target;
-    this.circuitBreaker = circuitBreaker;
+    this.egress = egress;
+    this.retryPolicyFunc = computeRetryPolicy(egress);
   }
 
-  @Override
-  public Future<HttpResponse<Buffer>> send(CloudEvent event) {
-    logger.debug("Sending event {} {}",
+  public Future<HttpResponse<Buffer>> send(final CloudEvent event) {
+    return send(event, 0);
+  }
+
+  private Future<HttpResponse<Buffer>> send(final CloudEvent event, final int retryCounter) {
+    logger.debug("Sending event {} {} {}",
       keyValue("id", event.getId()),
-      keyValue("subscriberURI", target)
+      keyValue("subscriberURI", target),
+      keyValue("retry", retryCounter)
     );
 
-    TracingSpan.decorateCurrentWithEvent(event);
 
-    final Future<HttpResponse<Buffer>> future = circuitBreaker.execute(breaker -> {
-      if (closed.get()) {
-        // Once sender is closed, return a successful future to avoid retrying.
-        breaker.tryComplete(null);
-        return;
-      }
+    final Promise<HttpResponse<Buffer>> promise = Promise.promise();
 
+    if (closed.get()) {
+      // Once sender is closed, return a successful future to avoid retrying.
+      promise.tryComplete(null);
+    } else {
       try {
+        TracingSpan.decorateCurrentWithEvent(event);
         requestEmitted();
-        send(event, breaker).
+        send(event, promise).
           onComplete(v -> requestCompleted());
       } catch (CloudEventRWException e) {
         logger.error("failed to write event to the request {}", keyValue("subscriberURI", target), e);
-        breaker.tryFail(e);
+        promise.tryFail(e);
       }
-    });
+    }
 
     // Handle closed return value.
-    return future.compose(r -> {
-      if (r == null) {
-       return Future.failedFuture("Sender closed for target="+target);
+    return promise.future().compose(
```

Comment:
I think at the end it's not that big a change, to have our own impl here, instead of a different dependency